### PR TITLE
Changes needed for `--cid-base` option in go-ipfs (simplified vesion)

### PR DESCRIPTION
**Pull request by kevina**
Simplified version of #8 

~Do not merge until https://github.com/ipfs/go-ipfs/pull/5789 is approved.~

### CODE REVIEW

**Comment by kevina** [comment]
@Stebalien when gx publishing this should it I use `minor` or `patch` I would think minor since it adds a new package.

**Review by magik6k — COMMENTED** [review]
Code LGTM, but it would be really nice to get some tests, especially for `--filter` in cid-fmt

**Review by Stebalien — COMMENTED** [review]
Same as @magik6k. LGTM but needs tests.

**Comment by kevina** [comment]
Okay I added tests and this should be good to go.  Will merge once I get an approving review.

**Comment by magik6k** [comment]
Can you enable CI here? Should be as simple as `mkdir ci; echo 'golang()' > ci/Jenkinsfile` + giving `ci` team on github write access (to set commit status)

File: cidenc/encoder.go
Comment (magik6k) [inline cidenc/encoder.go]:
```suggestion
// Encoder is a basic Encoder that will encode Cids using a specified
```

File: cidenc/encoder.go
Comment (magik6k) [inline cidenc/encoder.go]:
```suggestion
// Encode encodes the cid using the parameters of the Encoder
```

**Review by magik6k — APPROVED** [review]

**Comment by kevina** [comment]
Someone with the write permissions will need to activate Jenkins and Travis...

**Comment by Stebalien** [comment]
I've activated travis. Not sure how to activate Jenkins.

**Comment by Stebalien** [comment]
(I assume this was OK to merge... If not, we can revert it)

**Comment by kevina** [comment]
> I assume this was OK to merge... If not, we can revert it

No, not really, the commits to activate the CI's where untested and I was waiting for them to be activated before I merged this PR.  If that did not happen I would of removed the commit from the P.R. to do it separately.  In fact the Travis CI need some fixing.

Reverting will just make this messier I will create another P.R.

**Comment by magik6k** [comment]
> Not sure how to activate Jenkins

Just give `ci` team on GitHub write access to this repo

**Comment by Stebalien** [comment]
@magik6k done.

@kevina got it. Merged your fix as everything seems to work now.

**Comment by kevina** [comment]
@Stebalien This needs to be gx-published.  Should we bump the minor version or just the patch level?  I would think minor since it adds a new package.

**Comment by kevina** [comment]
> @Stebalien This needs to be gx-published. Should we bump the minor version or just the patch level? I would think minor since it adds a new package.

In the interest of pushing ipfs/go-ipfs#5789 forward, I  want ahead and gx published this as a minor release.

**Comment by Stebalien** [comment]
Either way, just don't bump to version 1.0.

For packages <= 1.0, I usually only bump the minor for breaking changes but that's mostly a habit from working with rust/js packages (https://nodesource.com/blog/semver-tilde-and-caret/#caretmajorzero).

**Pull request merged**
